discovery_engine/lib/src/infrastructure/type_adapters/hive_unique_id_adapter.dart

```diff
@@ -14,23 +14,37 @@
 
 import 'package:hive/hive.dart' show TypeAdapter, BinaryReader, BinaryWriter;
 import 'package:xayn_discovery_engine/src/domain/models/unique_id.dart'
-    show DocumentId;
+    show UniqueId, DocumentId, StackId;
 import 'package:xayn_discovery_engine/src/domain/repository/type_id.dart'
-    show documentIdTypeId;
+    show documentIdTypeId, stackIdTypeId;
 
-class DocumentIdAdapter extends TypeAdapter<DocumentId> {
+abstract class _UniqueIdAdapter<T extends UniqueId> extends TypeAdapter<T> {
   @override
-  final typeId = documentIdTypeId;
+  void write(BinaryWriter writer, T obj) {
+    final bytes = obj.value.buffer.asUint8List();
+    writer.writeByteList(bytes);
+  }
 
   @override
-  DocumentId read(BinaryReader reader) {
+  T read(BinaryReader reader) {
     final bytes = reader.readByteList();
-    return DocumentId.fromBytes(bytes);
+    switch (typeId) {
+      case documentIdTypeId:
+        return DocumentId.fromBytes(bytes) as T;
+      case stackIdTypeId:
+        return StackId.fromBytes(bytes) as T;
+      default:
+        throw ArgumentError.value('unrecognized "typeId"');
+    }
   }
+}
 
+class DocumentIdAdapter extends _UniqueIdAdapter<DocumentId> {
   @override
-  void write(BinaryWriter writer, DocumentId obj) {
-    final bytes = obj.value.buffer.asUint8List();
-    writer.writeByteList(bytes);
-  }
+  final typeId = documentIdTypeId;
```

Comment:
I'm surprised that this does not trigger https://dart-lang.github.io/linter/lints/overridden_fields.html

Reply:
but it's not overriding a field, it's assigning it, as the base class defines a getter
```dart
abstract class TypeAdapter<T> {
  /// Called for type registration
  int get typeId;
}
```